java/javax/servlet/jsp/el/ImplicitObjectELResolver.java

```diff
@@ -44,31 +44,32 @@
  * @since JSP 2.1
  */
 public class ImplicitObjectELResolver extends ELResolver {
+    private enum Scope {
+        APPLICATION_SCOPE, COOKIE, HEADER, HEADER_VALUES, INIT_PARAM, PAGE_CONTEXT, PAGE_SCOPE, PARAM, PARAM_VALUES,
+        REQUEST_SCOPE, SESSION_SCOPE;
```

Comment:
Aren't the `_SCOPE` suffixes redundant?

Reply:
They are consistent with the property values, but certainly redundant.

Reply:
Yes, both is true.